optimum/intel/openvino/modeling_decoder.py

```diff
@@ -240,32 +240,105 @@ def forward(
 
     # Adapted from transformers.models.gpt2.modeling_gpt2.GPT2LMHeadModel.prepare_inputs_for_generation
     def prepare_inputs_for_generation(self, input_ids, past_key_values=None, **kwargs):
-        attention_mask = kwargs.get("attention_mask", None)
+        past_key_values = past_key_values or kwargs.get("past", None)
+
+        # `past_key_values` may be in the stardard format (e.g. in contrastive search), converts to bloom's format if needed
+        if past_key_values is not None and self.config.model_type == "bloom":
+            if past_key_values[0][0].shape[0] == input_ids.shape[0]:
+                past_key_values = self._convert_to_bloom_cache(past_key_values)
 
         return {
             "input_ids": input_ids,
-            "past_key_values": past_key_values or kwargs.get("past", None),
+            "past_key_values": past_key_values,
             "use_cache": self.use_cache,
             "position_ids": None,
-            "attention_mask": attention_mask,
+            "attention_mask": kwargs.get("attention_mask", None),
             "token_type_ids": None,
         }
 
-    # Copied from transformers.models.gpt2.modeling_gpt2.GPT2LMHeadModel._reorder_cache
-    @staticmethod
     def _reorder_cache(
-        past_key_values: Tuple[Tuple[torch.Tensor]], beam_idx: torch.Tensor
+        self, past_key_values: Tuple[Tuple[torch.Tensor]], beam_idx: torch.Tensor
     ) -> Tuple[Tuple[torch.Tensor]]:
         """
         This function is used to re-order the `past_key_values` cache if [`~PreTrainedModel.beam_search`] or
-        [`~PreTrainedModel.beam_sample`] is called. This is required to match `past_key_values` with the correct
-        beam_idx at every generation step.
+        [`~PreTrainedModel.beam_sample`] is called.
+        This is required to match `past_key_values` with the correct beam_idx at every generation step.
         """
+        if self.config.model_type == "bloom":
```

Comment:
Is this the only case that has such a specific?

Reply:
Yes, bloom models needs the `past_key_values` to be reshaped before each generation steps to match the ones from the other architectures